gammapy/spectrum/models.py

```diff
@@ -26,6 +27,110 @@ def __call__(self, energy):
         """Call evaluate method of derived classes"""
         return self.evaluate(energy, **self.parameters)
 
+    def __str__(self):
+        """String representation"""
+        ss = self.__class__.__name__
+        for parname, parval in self.parameters.items():
+            ss += '\n{parname} : {parval:.3g}'.format(**locals())
+        return ss
+
+    def to_sherpa(self, name='default'):
+        """Return `~sherpa.models.ArithmeticModel`
+
+        Parameters
+        ----------
+        name : str, optional
+            Name of the sherpa model instance
+        """
+        import sherpa.models as m
+        if isinstance(self, PowerLaw):
+            model = m.PowLaw1D('powlaw1d.' + name)
+            model.gamma = self.parameters.index.value
+        else:
+            raise NotImplementedError
+
+        model.ref = self.parameters.reference.to('keV').value
+        model.ampl = self.parameters.amplitude.to('cm-2 s-1 keV-1').value
+
+        return model
+
+    @classmethod
+    def from_sherpa(cls, model):
```

Comment:
I'm +1 to move this logic that's specific for a given model to the class of that specific model. (see larger argument above concerning extensibility and maintainability).